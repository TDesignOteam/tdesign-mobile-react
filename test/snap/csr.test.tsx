import glob from 'glob';
import React from 'react';
import { render } from '@test/utils';
import { IGNORE_ASYNC_EXAMPLE_LIST } from './const';

function runTest() {
  const files = glob.sync('src/**/_example/*.jsx', {
    ignore: IGNORE_ASYNC_EXAMPLE_LIST,
  });

  describe('csr snapshot test', () => {
    files.forEach((file) => {
      it(`csr test ${file}`, async () => {
        const demo = await import(`../../${file}`);
        const RealDemoComp = demo.default ? demo.default : demo;
        const wrapper = render(<RealDemoComp />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
}

runTest();
