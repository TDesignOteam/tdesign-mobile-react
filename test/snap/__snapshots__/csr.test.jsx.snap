// Vitest Snapshot v1

exports[`csr snapshot test > csr test src/grid/_example/badge.tsx 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="t-grid"
      >
        <div
          class="t-grid-item"
          style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
        >
          <div
            class="t-badge"
          >
            <div
              class="t-badge__inner t-badge--dot t-badge--has-children t-badge--medium"
            />
            <img
              class="img-4 img-vertical"
              src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
            />
          </div>
          <div
            class="t-grid-item__text"
            style="padding-left: 0px;"
          >
            <div
              class="t-grid-item__title"
              style="padding-top: 8px; margin-bottom: 4px;"
            >
              <div
                class="text-4"
              >
                标题文字
              </div>
            </div>
            <div
              class="t-grid-item__description"
            />
          </div>
        </div>
        <div
          class="t-grid-item"
          style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
        >
          <div
            class="t-badge"
          >
            <div
              class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium"
            >
              8
            </div>
            <img
              class="img-4 img-vertical"
              src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
            />
          </div>
          <div
            class="t-grid-item__text"
            style="padding-left: 0px;"
          >
            <div
              class="t-grid-item__title"
              style="padding-top: 8px; margin-bottom: 4px;"
            >
              <div
                class="text-4"
              >
                标题文字
              </div>
            </div>
            <div
              class="t-grid-item__description"
            />
          </div>
        </div>
        <div
          class="t-grid-item"
          style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
        >
          <div
            class="t-badge"
          >
            <div
              class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium"
            >
              new
            </div>
            <img
              class="img-4 img-vertical"
              src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
            />
          </div>
          <div
            class="t-grid-item__text"
            style="padding-left: 0px;"
          >
            <div
              class="t-grid-item__title"
              style="padding-top: 8px; margin-bottom: 4px;"
            >
              <div
                class="text-4"
              >
                标题文字
              </div>
            </div>
            <div
              class="t-grid-item__description"
            />
          </div>
        </div>
        <div
          class="t-grid-item"
          style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
        >
          <div
            class="t-badge"
          >
            <div
              class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium"
            >
              ···
            </div>
            <img
              class="img-4 img-vertical"
              src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
            />
          </div>
          <div
            class="t-grid-item__text"
            style="padding-left: 0px;"
          >
            <div
              class="t-grid-item__title"
              style="padding-top: 8px; margin-bottom: 4px;"
            >
              <div
                class="text-4"
              >
                标题文字
              </div>
            </div>
            <div
              class="t-grid-item__description"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="t-grid"
    >
      <div
        class="t-grid-item"
        style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--dot t-badge--has-children t-badge--medium"
          />
          <img
            class="img-4 img-vertical"
            src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
          />
        </div>
        <div
          class="t-grid-item__text"
          style="padding-left: 0px;"
        >
          <div
            class="t-grid-item__title"
            style="padding-top: 8px; margin-bottom: 4px;"
          >
            <div
              class="text-4"
            >
              标题文字
            </div>
          </div>
          <div
            class="t-grid-item__description"
          />
        </div>
      </div>
      <div
        class="t-grid-item"
        style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium"
          >
            8
          </div>
          <img
            class="img-4 img-vertical"
            src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
          />
        </div>
        <div
          class="t-grid-item__text"
          style="padding-left: 0px;"
        >
          <div
            class="t-grid-item__title"
            style="padding-top: 8px; margin-bottom: 4px;"
          >
            <div
              class="text-4"
            >
              标题文字
            </div>
          </div>
          <div
            class="t-grid-item__description"
          />
        </div>
      </div>
      <div
        class="t-grid-item"
        style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium"
          >
            new
          </div>
          <img
            class="img-4 img-vertical"
            src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
          />
        </div>
        <div
          class="t-grid-item__text"
          style="padding-left: 0px;"
        >
          <div
            class="t-grid-item__title"
            style="padding-top: 8px; margin-bottom: 4px;"
          >
            <div
              class="text-4"
            >
              标题文字
            </div>
          </div>
          <div
            class="t-grid-item__description"
          />
        </div>
      </div>
      <div
        class="t-grid-item"
        style="flex-basis: 25%; flex-direction: column; padding-left: 0px; padding-right: 0px; align-items: center; justify-content: center; text-align: center;"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium"
          >
            ···
          </div>
          <img
            class="img-4 img-vertical"
            src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"
          />
        </div>
        <div
          class="t-grid-item__text"
          style="padding-left: 0px;"
        >
          <div
            class="t-grid-item__title"
            style="padding-top: 8px; margin-bottom: 4px;"
          >
            <div
              class="text-4"
            >
              标题文字
            </div>
          </div>
          <div
            class="t-grid-item__description"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ssr snapshot test > ssr test src/grid/_example/badge.tsx 1`] = `"<div class=\\"t-grid\\"><div class=\\"t-grid-item\\" style=\\"flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center\\"><div class=\\"t-badge\\"><div class=\\"t-badge__inner t-badge--dot t-badge--has-children t-badge--medium\\"></div><img class=\\"img-4 img-vertical\\" src=\\"https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png\\"/></div><div class=\\"t-grid-item__text\\" style=\\"padding-left:0\\"><div class=\\"t-grid-item__title\\" style=\\"padding-top:8px;margin-bottom:4px\\"><div class=\\"text-4\\">标题文字</div></div><div class=\\"t-grid-item__description\\"></div></div></div><div class=\\"t-grid-item\\" style=\\"flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center\\"><div class=\\"t-badge\\"><div class=\\"t-badge__inner t-badge--circle t-badge--has-children t-badge--medium\\">8</div><img class=\\"img-4 img-vertical\\" src=\\"https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png\\"/></div><div class=\\"t-grid-item__text\\" style=\\"padding-left:0\\"><div class=\\"t-grid-item__title\\" style=\\"padding-top:8px;margin-bottom:4px\\"><div class=\\"text-4\\">标题文字</div></div><div class=\\"t-grid-item__description\\"></div></div></div><div class=\\"t-grid-item\\" style=\\"flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center\\"><div class=\\"t-badge\\"><div class=\\"t-badge__inner t-badge--circle t-badge--has-children t-badge--medium\\">new</div><img class=\\"img-4 img-vertical\\" src=\\"https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png\\"/></div><div class=\\"t-grid-item__text\\" style=\\"padding-left:0\\"><div class=\\"t-grid-item__title\\" style=\\"padding-top:8px;margin-bottom:4px\\"><div class=\\"text-4\\">标题文字</div></div><div class=\\"t-grid-item__description\\"></div></div></div><div class=\\"t-grid-item\\" style=\\"flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center\\"><div class=\\"t-badge\\"><div class=\\"t-badge__inner t-badge--circle t-badge--has-children t-badge--medium\\">···</div><img class=\\"img-4 img-vertical\\" src=\\"https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png\\"/></div><div class=\\"t-grid-item__text\\" style=\\"padding-left:0\\"><div class=\\"t-grid-item__title\\" style=\\"padding-top:8px;margin-bottom:4px\\"><div class=\\"text-4\\">标题文字</div></div><div class=\\"t-grid-item__description\\"></div></div></div></div>"`;
